synnax/pkg/api/user.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2023 Synnax Labs, Inc.
+//
+// Use of this software is governed by the Business Source License included in the file
+// licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with the Business Source
+// License, use of this software will be governed by the Apache License, Version 2.0,
+// included in the file licenses/APL.txt.
+
+package api
+
+import (
+	"context"
+	"github.com/synnaxlabs/synnax/pkg/access"
+	"github.com/synnaxlabs/synnax/pkg/distribution/ontology"
+	"go/types"
+
+	"github.com/synnaxlabs/synnax/pkg/auth"
+	"github.com/synnaxlabs/synnax/pkg/auth/password"
+	"github.com/synnaxlabs/synnax/pkg/user"
+	"github.com/synnaxlabs/x/gorp"
+)
+
+// UserService is the core authentication service for the delta API.
+type UserService struct {
+	dbProvider
+	authProvider
+	userProvider
+	accessProvider
+}
+
+func NewUserService(p Provider) *UserService {
+	return &UserService{
+		dbProvider:     p.db,
+		authProvider:   p.auth,
+		userProvider:   p.user,
+		accessProvider: p.access,
+	}
+}
+
+// RegistrationRequest is an API request to register a new user.
+type RegistrationRequest struct {
+	auth.InsecureCredentials
+}
+
+// Register registers a new user with the provided credentials. If successful, returns a
+// response containing a valid JWT along with the user's details.
+func (s *UserService) Register(ctx context.Context, req RegistrationRequest) (tr TokenResponse, err error) {
+	return tr, s.WithTx(ctx, func(txn gorp.Tx) error {
+		if err := s.authenticator.NewWriter(txn).Register(ctx, req.InsecureCredentials); err != nil {
```

Comment:
Where are the access control checks for this method? Please make sure all relevant endpoints are access controlled, or we're running a security risk